test/health_test.go

```diff
@@ -123,8 +124,10 @@ func TestHealthMetrics(t *testing.T) {
 
 	cgateway.MetricsWaitGroup.Wait()
 
+	// sleep to avoid race conditions with m3.Client
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Don't feel good about using timer to get around race conditions, is there no other solution?